main/EDInitMod.F90

```diff
@@ -839,151 +843,174 @@ subroutine init_cohorts( site_in, patch_in, bc_in)
 
              else
                 temp_cohort%hite        = EDPftvarcon_inst%hgt_min(pft)
-                
                 ! Calculate the plant diameter from height
                 call h2d_allom(temp_cohort%hite,pft,temp_cohort%dbh)
 
-                ! Calculate the leaf biomass from allometry
-                ! (calculates a maximum first, then applies canopy trim)
-                call bleaf(temp_cohort%dbh,pft,temp_cohort%crowndamage, &
-                     temp_cohort%canopy_trim,c_leaf)
-             end if  ! sp mode
+             endif  ! sp mode
 
-             ! Calculate total above-ground biomass from allometry
-             call bagw_allom(temp_cohort%dbh,pft,temp_cohort%crowndamage,c_agw)
+          else ! interpret as initial diameter and calculate density 
+             if(hlm_use_nocomp .eq. itrue)then
+                temp_cohort%dbh = abs(EDPftvarcon_inst%initd(pft))
 
-             ! Calculate coarse root biomass from allometry
-             call bbgw_allom(temp_cohort%dbh,pft,c_bgw)
+                ! calculate crown area of a single plant
+                dummy_n = 1.0_r8 ! make n=1 to get area of one tree
+                spread = 1.0_r8  ! fix this to 0 to remove dynamics of canopy clousre, assuming a closed canopy. 
```

Comment:
There are these two named constants that you can optionally use for spread, if you like. Seems like we initialize with a spread of 0 for an inventory init, 1 for a bare ground init.
https://github.com/NGEET/fates/blob/sci.1.64.2_api.25.2.0/main/EDTypesMod.F90#L108-L109

Reply:
Looks like I should be using init_spread_inventory since this isn't a near bare ground case. I've changed it now.